File: API/Models/DTOs/GuardianDTO.cs
using System;

namespace API.Models.DTOs;

public class GuardianDTO
{
    public required string FirstName { get; set; }
    public string? SecondName { get; set; }
    public required string LastName { get; set; }
    public required string Gender { get; set; }
    public required string Relationship { get; set; }
    public required string IdentityNumber { get; set; }
    public string? Email { get; set; }
    public required string Phone { get; set; }
    public required DateOnly DateOfBirth { get; set; }
    public required AddressDTO AddressDTO { get; set; }
}

File: Client/src/app/_interfaces/guardian-dto.ts
import { AddressDto } from './address-dto';

export interface GuardianDto {
  firstName: string;
  secondName?: string;
  lastName: string;
  gender: string;
  relationship: string;
  identityNumber: string;
  email?: string;
  phone: string;
  dateOfBirth: string;
  AddressDto: AddressDto;
}
